service/storage_service.cc

```diff
@@ -5975,7 +5975,7 @@ future<locator::load_stats> storage_service::load_stats_for_tablet_based_tables(
                     return true;
                 }
 
-                bool is_pending = transition->pending_replica == me;
+                bool is_pending = *transition->pending_replica == me;
```

Comment:
is pending_replica ensured to be engaged here?

Reply:
I think this whole patch can be dropped. Disengaged optionals have operator== overloaded for the wrapped type peer, so this check as it was before the patch is correct and doesn't need to be changed

Reply:
@xemul is right, I just removed the patch and the code is still working.